File: temporal/worker-go/activities/child_activities.go
package activities

import (
	"context"
	"fmt"
	"time"

	"go.temporal.io/sdk/activity"
)

// ProcessItem simulates processing a single item in a fanout.
func ProcessItem(ctx context.Context, itemID string) (string, error) {
	logger := activity.GetLogger(ctx)
	logger.Info("ProcessItem", "itemId", itemID)

	// Simulate variable processing time (1-3 seconds)
	info := activity.GetInfo(ctx)
	processingTime := time.Duration(1000+info.Attempt*500) * time.Millisecond
	time.Sleep(processingTime)

	logger.Info(fmt.Sprintf("✅ Item processed: %s", itemID))
	return fmt.Sprintf("processed:%s", itemID), nil
}

File: temporal/worker-ts/src/activities/childActivities.ts
import { log } from '@temporalio/activity';

function delay(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

/**
 * processItem — simulates processing a single item with variable duration.
 */
export async function processItem(itemId: string): Promise<string> {
  log.info('processItem', { itemId });

  // Simulate variable processing time (1-3 seconds)
  const processingTime = 1000 + Math.floor(Math.random() * 2000);
  await delay(processingTime);

  log.info(`✅ Item processed: ${itemId} (${processingTime}ms)`);
  return `processed:${itemId}`;
}
